SC2039 wrongly trigged on $'..' C-like strings

#### For bugs
- Rule Id: SC2039
- My shellcheck version: online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env sh
printf 
```

#### Here's what shellcheck currently says:
```
Line 2	SC2039: In POSIX sh,  $'..'
```


#### Here's what shellcheck currently says:
```
Line 2:
printf $'%s\n' 'Hello world!'
       ^-- SC2039: In POSIX sh, $'..' is undefined.
```
#### Here's what I wanted or expected to see:

shellcheck should not complain on C-like strings `$'..'` because it is an accepted POSIX standard, and is supported by `ash`, `dash`, `ksh93`, `zsh` and probably a whole lot if not all modern POSIX-compliant shells.

#### Other legitimate use:

```sh
#!/usr/bin/env sh

IFS='$\n'
```

See: [SC2141](https://github.com/koalaman/shellcheck/wiki/SC2141)

The change has been applied to the working version of the standard: https://www.mail-archive.com/austin-group-l@opengroup.org/msg10365.html .
The warning that shellcheck produces has changed to SC3003 since the bug creation.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@
 ### Fixed
 
 ### Removed
+- SC3003: removed since ANSI C string is specified in POSIX.1-2024
 
 
 ## v0.11.0 - 2025-08-03

--- a/src/ShellCheck/Checks/ShellSupport.hs
+++ b/src/ShellCheck/Checks/ShellSupport.hs
@@ -213,7 +213,7 @@ prop_checkBashisms118 = verify checkBashisms "#!/bin/busybox sh\nxyz=1\n${!x*}" 
 prop_checkBashisms119 = verify checkBashisms "#!/bin/busybox sh\nx='test'\n${x^^[t]}" -- SC3059
 prop_checkBashisms120 = verify checkBashisms "#!/bin/sh\n[ x == y ]"
 prop_checkBashisms121 = verifyNot checkBashisms "#!/bin/sh\n# shellcheck shell=busybox\n[ x == y ]"
-prop_checkBashisms122 = verify checkBashisms "#!/bin/dash\n$'a'"
+prop_checkBashisms122 = verifyNot checkBashisms "#!/bin/dash\n$'a'"
 prop_checkBashisms123 = verifyNot checkBashisms "#!/bin/busybox sh\n$'a'"
 prop_checkBashisms124 = verify checkBashisms "#!/bin/dash\ntype -p test"
 prop_checkBashisms125 = verifyNot checkBashisms "#!/bin/busybox sh\ntype -p test"
@@ -240,8 +240,6 @@ checkBashisms = ForShell [Sh, Dash, BusyboxSh] $ \t -> do
 
     bashism (T_ProcSub id _ _) = warnMsg id 3001 "process substitution is"
     bashism (T_Extglob id _ _) = warnMsg id 3002 "extglob is"
-    bashism (T_DollarSingleQuoted id _) =
-        unless isBusyboxSh $ warnMsg id 3003 "$'..' is"
     bashism (T_DollarDoubleQuoted id _) = warnMsg id 3004 "$\"..\" is"
     bashism (T_ForArithmetic id _ _ _ _) = warnMsg id 3005 "arithmetic for loops are"
     bashism (T_Arithmetic id _) = warnMsg id 3006 "standalone ((..)) is"

